cli/cli-improvements.md

```diff
@@ -0,0 +1,315 @@
+Setup application in the current directory?   
+  >  1. Yes                                   
+    2. No
+
+nithya@Nithyas-MacBook-Pro bug7052 % rad init
+                                                         
+Initializing Radius. This may take a minute or two...    
+                                                         
+✅ Use existing Radius 0.30.0 install on radius-agent-aks
+✅ Use existing environment default                      
+✅ Scaffold application bug7052                          
+✅ Update local configuration                            
+ ```
+
+Key points that happen behind the scene:
+
+1. If it doesn't already exist, create a "default" environment.
+2. If it doesn't already exist, create a "default" resource group.
+3. Scaffold a default application, using the current directory's name.
+4. Update config.yaml and rad.yaml.
+
+This will be modified to:
+
+1. Request the name of the environment from the user and create an environment with that name under all-environments resource group
+2. Request user for application name.
+3. PUT group ```/planes/radius/local/resourceGroups/<app-name>-<env-name>```
+4. PUT application as ```/planes/radius/local/resourceGroups/<app-name>-<env-name>/providers/Applications.Core/applications/<app-name>```
+5. Continue to scaffold the application, using the current directory's name.
+6. Update both config.yaml and rad.yaml. No scope will be stored in any configuration file.
+
+
+#### New flow:
+
+```
+nithya@Nithyas-MacBook-Pro bug7052 % rad init
+
+Enter the name of your default environment: [staging] 
+                                              
+Setup application in the current directory?   
+  >  1. Yes                                   
+    2. No
+
+nithya@Nithyas-MacBook-Pro cool-app % rad init
+                                                         
+Initializing Radius. This may take a minute or two...    
+                                                         
+✅ Use existing Radius 0.30.0 install on radius-agent-aks
+✅ Use existing environment staging                      
+✅ Scaffold application cool-app in resource group cool-app-staging                         
+✅ Update local configuration                            
+ ```
+
+
+### rad group 
+
+The ```rad group``` family of commands have no changes. ```rad group switch``` alone will become obsolete since it deals with setting the current scope as a configuation in config.yaml.
+
+### rad env 
+
+The ```rad env``` family of commands currently operate in the scope that is stored in config.yaml, unless a scope is explicitly specified using -g flag.   
+
+Since the idea of "default" scope is going away, all rad env commands will need a mandatory -g flag to specify the group.
+
+#### rad env list 
+
+This could operate at two levels
+1. list all environments in a specified resource group
+```rad env list -g cool-group```
+2. list all environments across resource groups
+```rad env list```
```

Comment:
We will have a choice to make here between consistency and simplicity. If each environment is in its own group (by recommendation), then it's tricky how best to design this command.